src/odemis/acq/fastem.py

```diff
@@ -281,6 +284,9 @@ def __init__(self, scanner, multibeam, descanner, detector, stage, scan_stage, c
         self._settings_obs = settings_obs
         self._spot_grid_thresh = spot_grid_thresh
         self._blank_beam = blank_beam
+        self._total_roa_time = 0
+        if isinstance(future, model.ProgressiveFuture):
```

Comment:
Is this ever not a ProgressiveFuture?

Reply:
In the unittest cases the future can be `None` or `Mock()`, that's why I added this check